wp-includes/class-wp-web-app-manifest.php

```diff
@@ -155,6 +179,87 @@ public function get_manifest() {
 		return apply_filters( 'web_app_manifest', $manifest );
 	}
 
+	/**
+	 * Register test for lacking short_name in web app manifest.
+	 *
+	 * @since 0.4
+	 *
+	 * @param array $tests Tests.
+	 * @return array Tests.
+	 */
+	public function add_short_name_site_status_test( $tests ) {
+		$tests['direct']['web_app_manifest_short_name'] = array(
+			'label' => __( 'Short Name in Web App Manifest', 'pwa' ),
+			'test'  => array( $this, 'test_short_name_present_in_manifest' ),
+		);
+		return $tests;
+	}
+
+	/**
+	 * Test that web app manifest contains a short_name.
+	 *
+	 * @since 0.4
+	 * @todo Add test for PNG site icon.
+	 *
+	 * @return array Test results.
+	 */
+	public function test_short_name_present_in_manifest() {
+		$manifest = $this->get_manifest();
+
+		/* translators: %d is the max length as a number */
+		$description = sprintf( __( 'The <code>short_name</code> is a short version of your website&#8217;s name which is displayed when there is not enough space for the full name, for example with the site icon on a phone&#8217;s homescreen. It should be a maximum of %d characters long.', 'pwa' ), self::SHORT_NAME_MAX_LENGTH );
```

Comment:
I'd use a placeholder for `short_name`. Also, the first sentence is very long. I'd split it up to make translators' lives easier.